### PR TITLE
CALCITE-4646 Update to Jetty 9.4.42.v20210604

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

**Comment by stoty** [comment]
The failing Windows test may be caused by the new Jetty version.
This patch doesn't touch anything in the Avatica codebase that could case that.

**Comment by stoty** [comment]
The test fixes are for removing an illegal leading slash in the file paths on Windows.
These were accepted by earlier Jetty versions, but not by 9.4.42.

File: server/src/test/java/org/apache/calcite/avatica/server/BasicAuthHttpServerTest.java [hunk 1]
```diff
@@ -44,8 +44,8 @@
   private static String url;
 
   @BeforeClass public static void startServer() throws Exception {
-    final String userPropertiesFile = URLDecoder.decode(BasicAuthHttpServerTest.class
-        .getResource("/auth-users.properties").getFile(), "UTF-8");
+    final String userPropertiesFile = (new File(BasicAuthHttpServerTest.class
+        .getResource("/auth-users.properties").toURI())).getPath();
```
Comment (vlsi) [inline server/src/test/java/org/apache/calcite/avatica/server/BasicAuthHttpServerTest.java:48]:
```suggestion
    final String userPropertiesFile = Sources.of(BasicAuthHttpServerTest.class.getResource("/auth-users.properties"))
        .file().getAbsolutePath();
```

Please use `org.apache.calcite.util.Sources`

Comment (stoty) [reply server/src/test/java/org/apache/calcite/avatica/server/BasicAuthHttpServerTest.java:48]: @vlsi 
AFAIK Avatica doesn't depend on Calcite by design, and org.apache.calcite.util.Sources is not part of Avatica.

Would you like me to copy that class into Avatica, and use it ?

Comment (vlsi) [reply server/src/test/java/org/apache/calcite/avatica/server/BasicAuthHttpServerTest.java:48]: Oh, I missed it is `avatica`.
I think it makes sense to copy the class or something like that.

I do remember there is a lot of edge cases near File vs URL vs URI, so I would like to have a single class for conversion between URL and Files.

Comment (stoty) [reply server/src/test/java/org/apache/calcite/avatica/server/BasicAuthHttpServerTest.java]: I have copied most of Sources.java and Source.java to Avatica.
I've put them in core, but it may also make sense to put them in server or even server/tests.

File: server/src/test/java/org/apache/calcite/avatica/server/DigestAuthHttpServerTest.java [hunk 2]
```diff
@@ -44,8 +44,8 @@
   private static String url;
 
   @BeforeClass public static void startServer() throws Exception {
-    final String userPropertiesFile = URLDecoder.decode(BasicAuthHttpServerTest.class
-        .getResource("/auth-users.properties").getFile(), "UTF-8");
+    final String userPropertiesFile = (new File(BasicAuthHttpServerTest.class
+        .getResource("/auth-users.properties").toURI())).getPath();
```
Comment (vlsi) [inline server/src/test/java/org/apache/calcite/avatica/server/DigestAuthHttpServerTest.java:48]: Please use `Sources` here as well

File: server/src/test/java/org/apache/calcite/avatica/server/HttpAuthBase.java [hunk 3]
```diff
@@ -80,12 +80,11 @@ void readWriteData(String url, String tableName, Properties props) throws Except
 
   static String getHashLoginServicePropertiesString() {
     try {
-      final String userPropertiesFile =
-              URLDecoder.decode(HttpQueryStringParameterRemoteUserExtractorTest.class
-                      .getResource("/auth-users.properties").getFile(), "UTF-8");
+      final String userPropertiesFile = (new File(BasicAuthHttpServerTest.class
+          .getResource("/auth-users.properties").toURI())).getPath();
```
Comment (vlsi) [inline server/src/test/java/org/apache/calcite/avatica/server/HttpAuthBase.java:84]: `Sources` please

Comment (stoty) [reply server/src/test/java/org/apache/calcite/avatica/server/HttpAuthBase.java]: done

File: server/src/test/java/org/apache/calcite/avatica/server/DigestAuthHttpServerTest.java (hunk 2 above)
Comment (stoty) [reply server/src/test/java/org/apache/calcite/avatica/server/DigestAuthHttpServerTest.java]: done

**Review by zabetak — APPROVED** [review]
LGTM, if nobody objects I will merge this PR soon.
